bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSnapshotName.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc;
+
+import com.google.common.base.Preconditions;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class encapsulates a Snapshot name of the form
+ * projects/(projectId)/instances/(instanceId)/clusters/(clusterId)
+ */
+public class BigtableSnapshotName {
+  // Use a very loose pattern so we don't validate more strictly than the server.
+  private static final Pattern PATTERN =
+      Pattern.compile("projects/[^/]+/instances/([^/]+)/clusters/([^/]+)/snapshots/([^/]+)");
+
+
+  public static BigtableSnapshotName parse(String snapshotName) {
```

Comment:
Do we need a static factory method where the public constructor is doing the same thing?